shared/AppInsightsCommon/src/TelemetryItemCreator.ts

```diff
@@ -53,6 +53,9 @@ export function createTelemetryItem<T>(item: T,
     // Part C
     if (!isNullOrUndefined(customProperties)) {
         objForEachKey(customProperties, (prop, value) => {
+            if (Number.isNaN(value)) {
```

Comment:
While this would "work" for the example provided in issue, there are a couple of problems with this
- It's only going to support values directly assigned in the custom Properties (not any other field)
- If there is an "embedded" object then this won't detect / convert that either
- Calling the `Number.isNaN` function here will be expensive as function calls are expensive, it think that it would be "faster" to just use `if (value === NaN)` (we would need to do some perf testing).

We should probably review and do this within the channel Serializer.ts to better handle this, there is already some code in there for the measurements for handling `NaN` but the properties are expected to be strings...